src/features/gallery/GalleryPostActions.tsx

```diff
@@ -0,0 +1,83 @@
+import { IonIcon, IonRouterLink } from "@ionic/react";
+import { VoteButton } from "../post/shared/VoteButton";
+import { PostView } from "lemmy-js-client";
+import { chatbubbleOutline, shareOutline } from "ionicons/icons";
+import styled from "@emotion/styled";
+import { useAppSelector } from "../../store";
+import { useBuildGeneralBrowseLink } from "../../helpers/routes";
+import { getHandle } from "../../helpers/lemmy";
+import { css } from "@emotion/react";
+import MoreActions from "../post/shared/MoreActions";
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-around;
+  font-size: 1.5em;
+
+  max-width: 600px;
+  margin: auto;
+`;
+
+const Section = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 0.5rem;
+`;
+
+const Amount = styled.div`
+  font-size: 1rem;
+`;
+
+interface GalleryPostActionsProps {
+  post: PostView;
+  close: () => void;
+}
+
+export default function GalleryPostActions({
+  post,
+  close,
+}: GalleryPostActionsProps) {
+  const postVotesById = useAppSelector((state) => state.post.postVotesById);
+  const buildGeneralBrowseLink = useBuildGeneralBrowseLink();
+  const link = buildGeneralBrowseLink(
+    `/c/${getHandle(post.community)}/comments/${post.post.id}`
+  );
+
+  const score =
+    post.counts.score -
+    (post.my_vote ?? 0) +
+    (postVotesById[post.post.id] ?? 0);
```

Comment:
Is this duplicated from elsewhere? Should there be a helper for it?

Reply:
Should be much better now!